src/main/java/me/zeroeightsix/kami/module/modules/misc/HighwayTools.kt

```diff
@@ -0,0 +1,738 @@
+    private var buildWidth = register(Settings.integerBuilder("BuildWidth").withMinimum(1).withValue(7).withMaximum(9).withVisibility { page.value == Page.BUILD }.build())
+    private val rims = register(Settings.booleanBuilder("Rims").withValue(true).withVisibility { page.value == Page.BUILD }.build())
+    private var rimHeight = register(Settings.integerBuilder("RimHeight").withMinimum(1).withValue(1).withMaximum(clearHeight.value).withVisibility { rims.value && page.value == Page.BUILD }.build())
+    private val cornerBlock = register(Settings.booleanBuilder("CornerBlock").withValue(true).withVisibility { page.value == Page.BUILD }.build())
+
+    var ignoreBlocks = mutableListOf(ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "standing_sign")),
+            ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "wall_sign")),
+            ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "standing_banner")),
+            ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "wall_banner")),
+            ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "bedrock")),
+            ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "portal")))
+    var material: Block = ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "obsidian"))!!
+    var fillerMat: Block = ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "netherrack"))!!
+    private var playerHotbarSlot = -1
+    private var lastHotbarSlot = -1
+    private var isSneaking = false
+    var pathing = true
+    private var stuckDetector = 0
+    private var buildDirectionSaved = 0
+    private var buildDirectionCoordinate = 0
+    private val directions = listOf("North", "North-East", "East", "South-East", "South", "South-West", "West", "North-West")
+
+    //Stats
+    private var totalBlocksPlaced = 0
+    var totalBlocksDestroyed = 0
+    private var totalBlocksDistance = 0
+
+    //val blockQueue = PriorityQueue<BlockTask>(TaskComparator())
+    var blockQueue: Queue<BlockTask> = LinkedList<BlockTask>()
+    private val doneQueue: Queue<BlockTask> = LinkedList<BlockTask>()
+    private var blockOffsets = mutableListOf<Pair<BlockPos, Boolean>>()
+    private var waitTicks = 0
+    private var blocksPlaced = 0
+    private lateinit var currentBlockPos: BlockPos
+    private lateinit var startingBlockPos: BlockPos
+
+    fun isDone(): Boolean { return blockQueue.size == 0 }
+
+    override fun onEnable() {
+        if (mc.player == null) {
+            disable()
+            return
+        }
+        buildDirectionSaved = getPlayerDirection()
+        startingBlockPos = BlockPos(floor(mc.player.posX).toInt(), floor(mc.player.posY).toInt(), floor(mc.player.posZ).toInt())
+        currentBlockPos = startingBlockPos
+        playerHotbarSlot = mc.player.inventory.currentItem
+        lastHotbarSlot = -1
```

Comment:
if you do this in multiple places (not sure if you do), it would be cleaner to use a method

Reply:
i dont